src/views/v3/Bridge/AmountInput/index.tsx

```diff
@@ -0,0 +1,192 @@
+    value: string;
+    onChange: (event: string) => void;
+    onDebouncedChange: (event: string) => void;
+  }) => {
+    const [innerValue, setInnerValue] = useState<string>(value);
+    const [isFocused, setIsFocused] = useState(false);
+    const deferredOnChange = useDebouncedCallback(
+      onDebouncedChange,
+      INPUT_DEBOUNCE,
+    );
+
+    const onInnerChange: ChangeEventHandler<HTMLInputElement> = useCallback(
+      (e) => {
+        let value = e.target.value;
+        if (value === '.') value = '0.';
+
+        const numValue = Number(value);
+
+        if (isNaN(numValue) || numValue < 0) {
+          // allows all but negative numbers
+          return;
+        }
+
+        setInnerValue(e.target.value);
+        onChange(e.target.value); // callback with no delay
+        deferredOnChange(e.target.value);
+      },
+      [deferredOnChange, onChange],
+    );
+
+    // Propagate any outside changes to the inner TextField value
+    // The way we do this is by checking when the focus is not on the input component
+    useEffect(() => {
+      if (!isFocused) {
+        setInnerValue(value);
+      }
+      // We should run this sife-effect only when the value changes
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [value]);
+
+    return (
+      <TextField
+        {...props}
+        data-testid="amount-input"
+        value={innerValue}
+        focused={isFocused}
+        onChange={onInnerChange}
+        onFocus={() => setIsFocused(true)}
+        onBlur={() => setIsFocused(false)}
+      />
+    );
+  },
+);
+
+type Props = {
+  value: string;
+  debauncedValue: string;
+  supportedSourceTokens: Array<Token>;
+  tokenBalance: sdkAmount.Amount | null;
+  receiveAmount?: number | undefined;
+  error?: string;
+  warning?: string;
+  onChange: (value: string) => void;
+  onDebouncedChange: (value: string) => void;
+};
+
+/**
+ * Renders the input control to set the transaction amount
+ */
+const AmountInput = (props: Props) => {
+  const theme = useTheme();
+
+  const styles = useMemo(
+    () => ({
+      amountContainer: {
+        width: '100%',
+        maxWidth: '250px',
+      },
+      amountInput: {
+        background: theme.palette.input.background,
+        borderRadius: '8px',
+        border: 'none',
+      },
+      amountCardContent: {
+        display: 'flex',
+        alignItems: 'center',
+        height: '50px',
+        padding: 0,
+        ':last-child': {
+          padding: 0,
+        },
+      },
+      balance: {
+        color: theme.palette.text.secondary,
+        fontSize: '14px',
+        lineHeight: '14px',
+        textAlign: 'right',
+      },
+    }),
+    [theme],
+  );
+
+  const {
+    fromChain: sourceChain,
+    toChain: destChain,
+    isTransactionInProgress,
+  } = useSelector((state: RootState) => state.transferInput);
+
+  const { sourceToken, destToken } = useGetTokens();
+
+  const isInputDisabled = useMemo(
```

Comment:
Same as before, unnecessary useMemo. This in particular will add more of a performance penalty due to a function call over a simple boolean ||ing statement.